cairo-contracts/packages/clients/src/cometbft/component.cairo

```diff
@@ -470,6 +515,12 @@ pub mod CometClientComponent {
             self.client_sequence.write(client_sequence);
         }
 
+        fn write_update_height(
+            ref self: ComponentState<TContractState>, client_sequence: u64, update_height: Height
+        ) {
+            self.update_heights.entry(client_sequence).append().write(update_height)
```

Comment:
This does not order the update heights in ascending or descending order. So if there is an UpdateClient with height earlier than the latest height, it would cause the query to return incorrect result.

Reply:
Oh, that's absolutely right. I still haven't wired this `write_update_height` up anywhere. Meant to achieve a proper ordering with `Vec` yesterday but it's quite tricky as there is only a single `append` method available on `Vec`. I'll look into it more see what I can do.

Reply:
I ended up using the `Array` instead of `Vec` so can utilize a library available for sorting. This required me to implement the `Store` trait manually for `Array<Height>` to enable read and write operations once it’s placed in the `Map`. There is only a minor limitation (at least for now): the `Array` size can’t exceed 255.